# ================== Hyperparameter ================

# ------------ Allgemeine Einstellungen ------------
anzSamples: 300                    # anzahl der Samples

nurMotionModel: false              # wenn true --> zeigt nur das ausgewählte Motion Model an
                                   # ACHTUNG: wenn true --> es wird nicht resamplet

motionModelOdom: false             # decides which motionmodel should be taken
                                   # [true --> Odom; false --> Motion Command]    ACHTUNG: Problem bei Motion Model Odometry !!! --> funktioniert nicht einwandfrei (Problem beschreibung unten ab Zeile 58 und in Klasse von particle_filter.cpp)

normalTriangularDistribution: true # decides which distribution should be taken (!no significant change!)
                                   # [true --> normal distribution; false --> triangular distribution]

initialPosesNormalverteilt: true   # decides if the initial poses should be distributed
                                   # [true --> normal distribution; false --> all samples start at the initial pose]

sensorModelRangeFinder: true       # decides which sensormodel should be taken
                                   # [true --> range_finder; false --> Odom (In paper -> appraoch #3)]

clearingPoses: true                # poses werden vom letzen time-step in rviz gelöscht

# ------------ initiale Pose ------------
initialX: 0.5
initialY: 0.5
initialTH: 0.0

# ------------ Motion Models ------------
# --- sample_motion_model_velocity ---
alpha_1: 0.2 # 2.8
alpha_2: 0.8 # 3.4
alpha_3: 0.2 # 2.8
alpha_4: 0.8 # 3.4
alpha_5: 0.4 # 3.1
alpha_6: 0.6 # 3.2

# --- sample_motion_model_odometry ---
alpha_odom_1: 0.6
alpha_odom_2: 0.4
alpha_odom_3: 0.5
alpha_odom_4: 0.6

# ------------ Sensor Models ------------
laserScannerPositionX: 0.0 # Position des Laserscanners am Roboter in x-Richtung (bei Turtlebot Burger -> 0.0)
laserScannerPositionY: 0.0 # Position des Laserscanners am Roboter in y-Richtung (bei Turtlebot Burger -> 0.0)

jederWieVielteBeam: 20     # gibt an jeder wie vielte Beam des Scanners verwendet werden soll 

zHit: 0.5
zShort: 0.05
zMax: 3.5
zRand: 0.01
sigmaHit: 0.2 # 0.4 auch gut aber relativ aggressiv
              # 0.2 wie bei amcl
              # 0.8 nicht sehr aggressiv


# !!! Problembeschreibung Motion Model Odometry: !!!
# leider gibt es beim Motion model Odometry zwei Probleme:
#   - dieses Motion Model funktioniert nur jedes 3. Mal
#   - bei einem funktionierenden Versuch bewegen sich die Samples zwar in die richtige Richtung, jedoch ist die Bewegung viel kleiner als die des Roboters
#
#   da ich dieses Problm nicht lösen konnte habe ich das sample_motion_model_odometry auch nicht im Paper beschrieben,
#   wollte es jedoch nicht aus meinem Code löschen falls ich mir das Problem im Sommer nochmal anschaue oder Sie Feedback für mich haben
#   --> sample_motion_model_motion_command funktioniert einwandfrei
#       (optimale Hyperparameter zum testen siehe Paper)